### PR TITLE
doc/dev,.travis.yml,makefile: update testing docs

### LINKED ISSUE #811: Update `doc/dev/testing` docs
The developer docs in `doc/dev/testing` are out of date. They currently only talk about the go e2e test and running those tests using `go test ./test/e2e/...`, however we now have e2e tests for go, ansible, helm, and subcommands as well as unit and sanity tests. Everything is tested using make commands as well. The doc should be updated so developers know how to test their changes.

### PR DESCRIPTION
**Pull request by AlexNPavel**
**Description of the change:** Update docs in `doc/dev/testing`

**Motivation for the change:** See #811 

Closes #811

### CODE REVIEW

File: doc/dev/testing/running-the-tests.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+All other tests will run correctly on a remote cluster if `$HOME/.kube/config` points to the remote cluster and your
+`KUBECONFIG` env var is either empty or is set to the path of a kubeconfig for the remote cluster.
+
+## Cleanup of the Go E2E Tests
+
+The E2E tests create a new project using the operator-sdk to run in the provided
+cluster. The tests are designed to cleanup everything that gets created, but some errors
+during the go testscan cause these cleanups to fail (the ansible and helm E2E tests should
```
Comment (estroz) [inline doc/dev/testing/running-the-tests.md]:
```suggestion
during the go tests can cause these cleanups to fail (the ansible and helm E2E tests should
```

File: doc/dev/testing/running-the-tests.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+always clean up correctly). For example, if a segfault occurs or a user kills the
+testing process, the cleanup functions for the go tests will not run. To manually clean up a test:
+
+1. Delete the CRD (`kubectl delete -f $GOPATH/src/github.com/example-inc/memcached-operator/deploy/crds/cache_v1alpha1_memcached_crd.yaml`).
+2. Delete the created project in `$GOPATH/src/github.com/example-inc/memcached-operator`
+3. Delete the namespaces that the tests run in, which also deletes any resources created within the namespaces. The namespaces start with `memcached-memcached-group` or `main` and are appended with a unix timestamp (seconds since Jan 1 1970).
```
Comment (estroz) [inline doc/dev/testing/running-the-tests.md]: Give an example namespace deletion command.

File: doc/dev/testing/running-the-tests.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+# Running the SDK Tests
+
+The operator-sdk is automatically tested with a variety of tests anytime
```
Comment (LiliC) [inline doc/dev/testing/running-the-tests.md]: Nit: My personal preference is that we don't force a new line after x amount of characters for markdown.

File: doc/dev/testing/running-the-tests.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+## Running the Tests Locally
+
+To run the tests locally, the tests either need access to a remote kubernetes cluster or a
+local kubernetes instance running on the machine.
+
+### Remote Kubernetes Instance
+
+To run the tests on a remote cluster, the tests need access to a remote kubernetes cluster
+running kubernetes 1.11 as well as a docker image repo to push the operator image to,
```
Comment (shawn-hurley) [inline doc/dev/testing/running-the-tests.md]:
```suggestion
- running kubernetes 1.11 as well as a docker image repo to push the operator image to,
+ running kubernetes 1.11 or higher as well as docker image repo to push the operator image to,
```

File: doc/dev/testing/running-the-tests.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+### Local Minikube
+
+Another option for testing is using minikube. This is not advised as it uses vanilla kubernetes, which has less
+strict security and may allow some tests to pass when they would not under openshift. Minikube is faster than
+openshift and uses less RAM though. To start the minikube cluster, download and install the proper [binary][minikube-binary]
+for your system and run these commands.
+
+```sh
+$ minikube start
```
Comment (shawn-hurley) [inline doc/dev/testing/running-the-tests.md]: You may want to add the version flag here, you must use greater than 1.11 and the default is still 1.10 I think.

File: doc/dev/testing/running-the-tests.md (hunk 3 above)
Comment (AlexNPavel) [reply doc/dev/testing/running-the-tests.md]: I don't think it should be forced, but I feel that it makes it easier to review if the lines are split up more frequently.

File: doc/dev/testing/running-the-tests.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+To fully delete the cluster, you must run these commands:
+
+```sh
+$ oc cluster down
+$ sudo umount $(grep $HOME/oscluster /proc/mounts | cut -f2 -d" " | sort -r)
+$ sudo rm -rf $HOME/oscluster
+```
+
+**NOTE**: Starting with openshift 4.0, the `oc cluster up` command will not be availble. This document
```
Comment (LiliC) [inline doc/dev/testing/running-the-tests.md]:
```suggestion
 **NOTE**: Starting with openshift 4.0, the `oc cluster up` command will not be available. This document
```

**Pull request merged**